### PR TITLE
Tooltip characteristics, docs, custom styling

### PR DESCRIPTION
**Pull request by libertymayc**

### CODE REVIEW

File: packages/core/stories/tooltip.stories.tsx [hunk 1]
```diff
@@ -73,20 +98,67 @@ ScrollTooltip.args = {
   placement: "top",
 };
 
-export const ErrorTooltip: ComponentStory<typeof Tooltip> = Template.bind({});
```
Comment (joshwooding) [inline packages/core/stories/tooltip.stories.tsx:76]: What's the benefit on not using templates?

Comment (libertymayc) [reply packages/core/stories/tooltip.stories.tsx:76]: @joshwooding if you try putting all 3 tooltip stories into the docs page it crashes, this was to try and fix that (but it doesn't) - i believe its something to do with the tooltip hook, would you be able to take a look?

**Review by heswell — COMMENTED** [review]
MInor thing - in the custom example with icon and text, the text is hard against the icon, could use a little space

File: packages/core/src/tooltip/Tooltip.css [hunk 2]
```diff
@@ -1,39 +1,41 @@
-.uitk-density-touch .uitkTooltip {
+.uitk-density-touch {
   --icon-top: 6px;
```
Comment (alycrys) [inline packages/core/src/tooltip/Tooltip.css:2]: should this be '--tooltip-icon-top'?

File: packages/core/src/tooltip/Tooltip.css [hunk 3]
```diff
@@ -1,39 +1,41 @@
...
-  padding: var(--tooltip-padding);
+  border-color: var(--uitkTooltip-borderColor, var(--tooltip-status-borderColor));
+  border-width: var(--uitkTooltip-borderWidth, var(--uitk-container-borderWidth));
+  border-style: var(--uitkTooltip-borderStyle, var(--uitk-container-borderStyle));
+  box-shadow: var(--uitkTooltip-shadow, var(--uitk-overlayable-shadow-popout));
+  color: var(--uitkTooltip-text-color, var(--uitk-text-primary-foreground));
+  font-size: var(--tooltip-fontSize);
+  line-height: var(--tooltip-lineHeight);
+  max-width: var(--uitkTooltip-maxWidth, 230px);
+  min-height: calc(var(--tooltip-fontSize) * var(--tooltip-lineHeight));
```
Comment (alycrys) [inline packages/core/src/tooltip/Tooltip.css:35]: not sure if this formula is correct. what if they set '--uitkTooltip-lineHeight' in px?

File: tooling/css-variable-docgen-components/src/CharacteristicUsage.tsx [hunk 4]
```diff
@@ -53,7 +53,7 @@ export function CharacteristicUsage(
           {Object.entries(characteristicTokenMap)
             .sort((a, b) => a[0].localeCompare(b[0]))
             .map(([name, tokens]) => (
-              <CharacteristicUsageRow name={name} tokens={tokens} />
+              <CharacteristicUsageRow key={name} name={name} tokens={tokens} />
```
Comment (alycrys) [inline tooling/css-variable-docgen-components/src/CharacteristicUsage.tsx:56]: 👍

File: tooling/css-variable-docgen-components/src/CharacteristicUsageRow.tsx [hunk 5]
```diff
@@ -21,7 +21,7 @@ export const CharacteristicUsageRow: FC<CharacteristicUsageRowProps> = (
       <td>
         {tokens &&
           tokens.map((token) => (
-            <Description>
+            <Description key={token}>
```
Comment (alycrys) [inline tooling/css-variable-docgen-components/src/CharacteristicUsageRow.tsx:24]: 👍

File: packages/core/src/tooltip/Tooltip.tsx [hunk 6]
```diff
@@ -111,7 +111,7 @@ export const Tooltip = forwardRef<HTMLDivElement, TooltipProps>(
             ) : (
               <>
                 {getIcon({})}
-                <span className={withBaseName("body")}>{title}</span>
+                <span className={withBaseName("title")}>{title}</span>
```
Comment (joshwooding) [inline packages/core/src/tooltip/Tooltip.tsx:114]: Is there a reason for this name change other than preference? I think we used to call it a tooltip body but I don't have a preference in the name.

File: packages/core/src/tooltip/Tooltip.css [hunk 7]
```diff
@@ -1,39 +1,40 @@
...
 }
 
-.uitk-density-high .uitkTooltip {
-  --icon-top: 1px;
+.uitk-density-high {
+  --tooltip-icon-top: 1px;
 }
 
+/* Styles applied to root component */
 .uitkTooltip {
```
Comment (joshwooding) [inline packages/core/src/tooltip/Tooltip.css:18]: Not a change needed in this PR but we should standardise placement of css variables.

Should it go in a separate class, top of the old class, should we use the css var api inline vs creating a private token etc.

**Review by alycrys — APPROVED** [review]

**Review by alycrys — APPROVED** [review]

**Pull request closed without merging**